### ISSUE TITLE
[Vue] Combobox - Poor performance when scrolling in the list with a lot of options

### ISSUE THREAD

**Issue by klausXR** [tailwindlabs/headlessui#2527]
**What package within Headless UI are you using?**

@headlessui/vue

**What version of that package are you using?**

1.7.1, but the source code about the issue has not changed

**What browser are you using?**

Not important

**Describe your issue**

I'm using the Combobox component to show a list of countries with a checkbox inside each element ( there are about 100 options to choose from with no filter applied ), however, scrolling inside the body of the combobox is very janky.

Turned on the performance profiler to see what was going on and I noticed that mousemove events generate a lot of work for the cpu, because of their frequent calls.

Is there a reason why that event is preferred over mouseenter/mouseleave, which trigger way less frequently?

I know I can perhaps use some form of virtualization to maybe fix the issue, but I feel like 100 components should not cause an issue if everything is set up properly

**Comment by adi518** [comment]
By default, such components do not offer virtualization, so it's nothing to be surprised about. I'd use https://tanstack.com/virtual/v3 and pass a custom list component.

**Comment by klausXR** [comment]
Its the poor performance that surprises me, not the lack of virtualization

**Comment by adi518** [comment]
Oh, definitely sounds like an optimization bug then.

**Comment by voqse** [comment]
That is to be expected. Each Option creation involves several performance-intensive tasks, such as extensive DOM manipulation primarily for accessibility purposes. Consequently, I am hesitant to utilize this specific component, despite its thoughtful design.

**Comment by RobinMalfait** [comment]
Hey, thank you for this bug report! 🙏

This should be improved by #2574, and will be available in the next release. Feel free to try it out and let me know if this improves things for you and/or if you are experiencing other performance issues.

You can already try it using:
- `npm install @headlessui/react@insiders`.
- `npm install @headlessui/vue@insiders`.

**Comment by klausXR** [comment]
> Hey, thank you for this bug report! 🙏
> 
> This should be improved by #2574, and will be available in the next release. Feel free to try it out and let me know if this improves things for you and/or if you are experiencing other performance issues.
> 
> You can already try it using:
> 
> * `npm install @headlessui/vue@insiders`.

I'm not on my work computer currently, but I'll give it a test later tonight. I did see that the fix implemented was not in the direction that I proposed. While I did not dig deep enough to confirm that what I was thinking was indeed the problem, I would still ask, is there any difference for the sake of this component between mousemove and mouseenter events for the options? 

The mousemove event fires very frequently, so handlers must be right. While there's nothing heavy in the `handleMove` function, in the Vue component, it still reads reactive properties ( props.disabled and active.value ). Accessing variables is not necessarily something to worry about, but here proxies are involved and they trigger Vue's dependency tracking.

Perhaps this code solves an edge case with which I am not familiar, but in case it doesnt, mouseenter/leave events should be a safer option when performance is considered.

These things are often hard to measure on our top of the line development machines, but they reach end users eventually and may cause frustration. 

Pointing this out just in the case that there isn't an underlying issue being solved by that specific event. 

Thanks for the good work :)
